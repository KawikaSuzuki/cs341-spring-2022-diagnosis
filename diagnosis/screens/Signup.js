import React, {useState}  from 'react';
import { View, Text } from 'react-native';
import { StatusBar } from 'expo-status-bar';

//formik
import { Formik } from 'formik';

import { StyledContainer, 
    InnerContainer,
    PageTitle, 
    StyledForm, 
    LabelIcon, 
    Colors, 
    StyleTextInput, 
    EyeIcon,
    StyledButton,
    ButtonText,
    Line,
    ExtraView,
    ExtraText,
    TextLink,
    TextLinkContent,
    StyleInputLabel
} from "./../components/styles";

import {Octicons, Ionicons } from '@expo/vector-icons';

const {company, placeholder} = Colors;

//keyboard avoiding wrapper
import KeyboardAvoidWrap from '../components/KeyboardAvoidWrap';

const Signup = ({navigation}) => {
    const [hidePassword, setHidePassword] = useState(true);
    return (
        <KeyboardAvoidWrap>
            <StyledContainer>
                <StatusBar style="dark" />
                <InnerContainer>
                    <PageTitle>Sign Up to PDX PDX</PageTitle>
                    <Formik
                        initialValues={{fullName: '', email: '', username: '', password: '', confirmPassword: ''}}
                        onSubmit={(values) => {
                            console.log(values);
                        }}
                    >{({handleChange, handleBlur, handleSubmit, values}) => (
                        <StyledForm>
                            <TextInput 
                                label="Full Name"
                                icon="person"
                                placeholder="John Doe"
                                placeholderTextColor={placeholder}
                                onChangeText={handleChange('fullName')}
                                onBlur={handleBlur('fullName')}
                                value={values.fullName}
                            />
                            <TextInput 
                                label="Username *"
                                mode="outlined"
                                icon="person"
                                placeholder="username"
                                placeholderTextColor={placeholder}
                                onChangeText={handleChange('username')}
                                onBlur={handleBlur('username')}
                                value={values.username}
                            />
                            <TextInput 
                                label="Email Address *"
                                icon="mail"
                                placeholder="your-name@example.com"
                                placeholderTextColor={placeholder}
                                onChangeText={handleChange('email')}
                                onBlur={handleBlur('email')}
                                value={values.email}
                                keyboardType="email-address"
                            />
                            <TextInput 
                                label="Password *"
                                icon="lock"
                                placeholder="********"
                                placeholderTextColor={placeholder}
                                onChangeText={handleChange('password')}
                                onBlur={handleBlur('password')}
                                value={values.password}
                                secureTextEntry={hidePassword}
                                isPassword={true}
                                hidePassword={hidePassword}
                                setHidePassword={setHidePassword}
                            />
                            <TextInput 
                                label="Confirm Password *"
                                icon="lock"
                                placeholder="********"
                                placeholderTextColor={placeholder}
                                onChangeText={handleChange('confirmPassword')}
                                onBlur={handleBlur('confirmPassword')}
                                value={values.confirmPassword}
                                secureTextEntry={hidePassword}
                                isPassword={true}
                                hidePassword={hidePassword}
                                setHidePassword={setHidePassword}
                            />
                            <Text>{"\n"}</Text>
                            <StyledButton onPress={handleSubmit}>
                                <ButtonText>Sign Up</ButtonText>
                            </StyledButton>
                            <Line />
                            <ExtraView>
                                <ExtraText>Already have an account? </ExtraText>
                                <TextLink onPress={() => navigation.navigate("Login")}>
                                    <TextLinkContent>Login</TextLinkContent>
                                </TextLink>
                            </ExtraView>
                        </StyledForm>)}
                    </Formik>
                </InnerContainer>
            </StyledContainer>
        </KeyboardAvoidWrap>
    );
}

const TextInput = ({label, icon, isPassword, hidePassword, setHidePassword, ...props}) => {
    return (
        <View>
            <LabelIcon>
                <Octicons name={icon} size={30} color={company}/>
            </LabelIcon>
            <StyleInputLabel>{label}</StyleInputLabel>
            <StyleTextInput {...props} />
            {isPassword && (
                <EyeIcon onPress={() => setHidePassword(!hidePassword)}>
                    <Ionicons name={hidePassword ? 'md-eye-off' : 'md-eye'} size={30} color={placeholder} />
                </EyeIcon>
            )}
        </View>
    )
}

export default Signup;