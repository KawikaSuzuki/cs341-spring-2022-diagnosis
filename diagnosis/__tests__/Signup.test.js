import React from 'react';
import renderer from 'react-test-renderer';
import Signup from '../screens/Signup';

describe('<Signup />', () => {
    test("has 1 child", async () => {
        const tree = renderer.create(<Signup />).toJSON();
        await expect(tree.children.length).toBe(1);
    });

});