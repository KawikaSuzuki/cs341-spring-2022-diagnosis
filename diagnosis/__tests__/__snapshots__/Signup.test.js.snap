// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        accessible={true}
        className="sc-bdvvtL zxezy"
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          className="sc-gsDKAQ rZFaE"
        >
          <Text
            className="sc-hKwDye fAcRlN"
          >
            Sign Up to PDX PDX
          </Text>
          <View
            className="sc-eCImPb kMRaWM"
          >
            <View>
              <View
                className="sc-iCfMLu eAWtPN"
              >
                <Text />
              </View>
              <Text
                className="sc-gKclnd dsijCe"
              >
                Full Name
              </Text>
              <TextInput
                allowFontScaling={true}
                className="sc-jRQBWg fNWPqr"
                onBlur={[Function]}
                onChangeText={[Function]}
                placeholder="John Doe"
                placeholderTextColor="#ff000067"
                rejectResponderTermination={true}
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View>
              <View
                className="sc-iCfMLu eAWtPN"
              >
                <Text />
              </View>
              <Text
                className="sc-gKclnd dsijCe"
              >
                Username *
              </Text>
              <TextInput
                allowFontScaling={true}
                className="sc-jRQBWg fNWPqr"
                mode="outlined"
                onBlur={[Function]}
                onChangeText={[Function]}
                placeholder="username"
                placeholderTextColor="#ff000067"
                rejectResponderTermination={true}
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View>
              <View
                className="sc-iCfMLu eAWtPN"
              >
                <Text />
              </View>
              <Text
                className="sc-gKclnd dsijCe"
              >
                Email Address *
              </Text>
              <TextInput
                allowFontScaling={true}
                className="sc-jRQBWg fNWPqr"
                keyboardType="email-address"
                onBlur={[Function]}
                onChangeText={[Function]}
                placeholder="your-name@example.com"
                placeholderTextColor="#ff000067"
                rejectResponderTermination={true}
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View>
              <View
                className="sc-iCfMLu eAWtPN"
              >
                <Text />
              </View>
              <Text
                className="sc-gKclnd dsijCe"
              >
                Password *
              </Text>
              <TextInput
                allowFontScaling={true}
                className="sc-jRQBWg fNWPqr"
                onBlur={[Function]}
                onChangeText={[Function]}
                placeholder="********"
                placeholderTextColor="#ff000067"
                rejectResponderTermination={true}
                secureTextEntry={true}
                underlineColorAndroid="transparent"
                value=""
              />
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
            </View>
            <View>
              <View
                className="sc-iCfMLu eAWtPN"
              >
                <Text />
              </View>
              <Text
                className="sc-gKclnd dsijCe"
              >
                Confirm Password *
              </Text>
              <TextInput
                allowFontScaling={true}
                className="sc-jRQBWg fNWPqr"
                onBlur={[Function]}
                onChangeText={[Function]}
                placeholder="********"
                placeholderTextColor="#ff000067"
                rejectResponderTermination={true}
                secureTextEntry={true}
                underlineColorAndroid="transparent"
                value=""
              />
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                className="sc-jrQzAO dafyie"
              >
                Sign Up
              </Text>
            </View>
            <View
              className="sc-kDTinF glXNyz"
            />
            <View
              className="sc-iqseJM DBbKg"
            >
              <Text
                className="sc-crHmcD bLjgwf"
              >
                Already have an account? 
              </Text>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  className="sc-bqiRlB kAwtfN"
                >
                  Login
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
